src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java

```diff
@@ -662,12 +662,13 @@ private LdapContext bind(String principalName, String password, SocketInfo serve
             Thread.currentThread().setName("Connecting to "+ldapUrl+" : "+oldName);
             LOGGER.fine("Connecting to " + ldapUrl);
             try {
+                props.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
```

Comment:
Consistent with e.g. spring-projects/spring-ldap@0a1d3f595b670412b7fd7ab739237c57381a7fa8. Rather than directly importing this class, which is supposed to be referenced only from the `java.naming` module, we pass it in as a string. The `java.naming` module can then interpret that string as it sees fit internally.